web/src/lib/components/map/mapUtils.ts

```diff
@@ -0,0 +1,288 @@
+import Map from "ol/Map";
+import View from "ol/View";
+import SimpleGeometry from "ol/geom/SimpleGeometry";
+import { type Coordinate } from "ol/coordinate";
+import GeoJSON from "ol/format/GeoJSON";
+
+import { Vector as VectorSource, Cluster, OSM } from "ol/source";
+import { WebGLTile as TileLayer, Layer } from "ol/layer";
+import { fromLonLat } from "ol/proj";
+
+import WebGLVectorLayerRenderer from "ol/renderer/webgl/VectorLayer.js";
+import WebGLPointsLayer from "ol/layer/WebGLPoints.js";
+
+import { boundingExtent, type Extent } from "ol/extent";
+
+import { Circle, Fill, Icon, Stroke, Style, Text } from "ol/style.js";
+import { Vector as VectorLayer } from "ol/layer.js";
+
+import type { FeatureLike } from "ol/Feature";
+import type { Options as OptionsLayer } from "ol/layer/Layer";
+import type { VectorStyle } from "ol/render/webgl/VectorStyleRenderer";
+import type { WebGLStyle } from "ol/style/webgl";
+
+const docElement = document.documentElement;
+const style = getComputedStyle(docElement);
+
+/**
+ * Variables gotten from css vars.
+ */
+const cssVars = {
+	text_sm_semibold: style.getPropertyValue("--text-sm-semibold"),
+	indigo_400: style.getPropertyValue("--indigo-400"),
+};
+
+/**
+ * Default style for vector layer.
+ */
+const defaultVectorStyle: VectorStyle = {
+	"stroke-color": "#fff",
+	"fill-color": "#3980a895",
+};
+
+/**
+ * Default style for WebGl point layer.
+ */
+const defaultIconStyle: WebGLStyle = {
+	"icon-src": "/images/logo.svg",
+};
+
+/**
+ * Default style for cluster point layer.
+ */
+const defaultClusterIcon = new Style({
+	image: new Icon({
+		src: "/images/logo.svg",
+	}),
+});
+
+/**
+ * Default style for cluster symbol.
+ */
+const defaultClusterSymbol: Style = new Style({
+	image: new Circle({
+		radius: 20,
+		stroke: new Stroke({
+			color: cssVars.indigo_400,
+			width: 2,
+		}),
+		fill: new Fill({
+			color: "#fff",
+		}),
+	}),
+	text: new Text({
+		text: "",
+		font: cssVars.text_sm_semibold,
+		textAlign: "center",
+		fill: new Fill({
+			color: "#000",
+		}),
+	}),
+});
+
+/**
+ * Custom vector layer
```

Comment:
Instead of this docs, this can be refactored to `WebGL vector layer for Open Layers.`